csharp/src/Ice/Reference.cs

```diff
@@ -584,6 +584,101 @@ static void StartQueryOption(StringBuilder sb, ref bool firstOption)
             }
         }
 
+        internal async ValueTask<Connection> GetConnectionAsync(CancellationToken cancel)
+        {
+            (Connection? connection, bool cached, IReadOnlyList<Endpoint> endpoints, List<Connector>? connectors) =
+                await GetExistingConnectionAsync(cancel).ConfigureAwait(false);
+
+            if (connection == null)
+            {
+                OutgoingConnectionFactory connectionFactory = Communicator.OutgoingConnectionFactory;
+                if (connectors == null)
+                {
+                    (cached, endpoints) = await ComputeEndpointsAsync(cancel).ConfigureAwait(false);
+                    connectors = await connectionFactory.ComputeConnectorsAsync(
+                        this,
+                        endpoints,
+                        ImmutableList<Connector>.Empty,
+                        cancel).ConfigureAwait(false);
+                }
+
+                Debug.Assert(connectors.Count > 0);
+                for (int i = 0; i < connectors.Count && connection == null; ++i)
```

Comment:
Not a big deal but I find that a "break;" in the loop is clearer than checking for "connection == null" here.